src/dodal/devices/oav/oav_to_redis_forwarder.py

```diff
@@ -103,9 +103,9 @@ async def _confirm_mjpg_stream(self, response, stream_url):
 
     @AsyncStatus.wrap
     async def kickoff(self):
-        await self._open_connection_and_(self._confirm_mjpg_stream)
+        await self._open_connection_and_do_function(self._confirm_mjpg_stream)
         self.forwarding_task = asyncio.create_task(
```

Comment:
If we error between `kickoff` and `complete` then this task will never be cancelled. I think either this part of code or the plan should make sure the task is cancelled upon exceptions

Reply:
Good spot. I think if the RE fails it will call stop on all the devices and so by implementing that we fix this issue

Reply:
In doing this I also realised we're using cancel poorly, have refactored into a `stop` flag